dvc/repo/init.py

```diff
@@ -63,15 +63,21 @@ def init(root_dir=os.curdir, no_scm=False, force=False):
     Raises:
         KeyError: Raises an exception.
     """
+
+    if no_scm and subrepo:
+        raise InitError(
+            "Cannot initialize repo with `--no-scm` and `--subrepo`"
+        )
+
     root_dir = os.path.realpath(root_dir)
     dvc_dir = os.path.join(root_dir, Repo.DVC_DIR)
-    scm = SCM(root_dir)
+
+    scm = SCM(root_dir, search_parent_directories=subrepo)
     if isinstance(scm, NoSCM) and not no_scm:
         raise InitError(
             "{repo} is not tracked by any supported scm tool (e.g. git). "
-            "Use `--no-scm` if you don't want to use any scm.".format(
-                repo=root_dir
-            )
+            "Use `--no-scm` if you don't want to use any scm or "
+            "`--subrepo` if sub-repository is needed.".format(repo=root_dir)
```

Comment:
Something more descriptive like:
```
"'{repo}' is not tracked by Git."
"Use `--no-scm` if you don't want to use Git"
" or `--subrepo` if initializing inside a subdirectory of a parent DVC repository."
```
> Obvi rewarp

Reply:
Note: I thought we decided previously to just talk about Git in user output since we only support that, even when code is `scm` but if I'm wrong, just use "SCM" or "a SCM (e.g. Git)", etc. instead of "Git". Please use proper capitalization. Thanks